### ISSUE TITLE
Google Compute's google_compute_instance_template.network_interface.network should be a URL

### ISSUE THREAD

**Issue by sorccu** [hashicorp/terraform#3226]
On Terraform v0.6.3, it looks like `google_compute_instance_template.network_interface.network` is not supposed to be a network name like the [documentation suggests](https://www.terraform.io/docs/providers/google/r/compute_instance_template.html#network). You will get the following error:

```
* google_compute_instance_template.foo: Error creating instance: googleapi: Error 400: Invalid value for field 'resource.properties.networkInterfaces[0].network': 'bar'.  The URL is malformed., invalid
```

Using the network's `.self_link` works.

The "Example Usage" on that page should also be wrong, unless there is some sort of special handling for the "default" network.

**Comment by lwander** [comment]
Oddly enough, "default" works on its own. I think it would be neater to simply provide a `network` name rather than note in the documentation to provide a URI to the network. What do you think?

**Comment by sorccu** [comment]
I think it would be best to go for consistency, so that "default" isn't a special case. I'd prefer just using the network name.

**Comment by lwander** [comment]
Sounds good, I'll make the fix

**Comment by lwander** [comment]
@sorccu, @sparkprime : fixed by #3242
